datadog-integrations-aws-handler/src/datadog_integrations_aws/handlers.py

```diff
@@ -73,6 +75,16 @@ def create_handler(
                 message=f"Error creating AWS account: {e}",
                 errorCode=http_to_handler_error_code(e.status)
             )
+    if model.ExternalIDSecretName is not None:
+        secret_name = model.ExternalIDSecretName
+    else:
+        secret_name = DEFAULT_SECRET_NAME
+    boto_client = session.client("secretsmanager", "us-east-1")
```

Comment:
We'd need to detect the region the stack's in automatically I believe, I don't think we can always create in us-east-1

Reply:
Yeah, thanks for the catch. I'll update this.